### ISSUE TITLE
Flaky tests not displayed in 2.10.0

### ISSUE THREAD

**Issue by PavloNikolenko** [allure-framework/allure2#910]
**Describe the bug**
Flaky test "bomb" icon is no longer displayed in 2.10.0

**To Reproduce**
1. Run flaky test (test that fails on first attempt and passes on retry)
2. Generate report using allure 2.10.0

**Expected behavior**
Flaky tests would be marked with a "bomb" icon on the report

**Screenshots**
<img width="1080" alt="screen shot 2019-03-07 at 3 42 52 pm copy" src="https://user-images.githubusercontent.com/15851089/53997248-4aa27400-40f0-11e9-9fd7-7737cabe57b1.png">

**Environment:**

| Allure version | 2.10.0 |
| --- | --- |
| Test framework | cucumber 4.2.6 |
| Allure adaptor | allure-cucumber4-jvm 2.10.0 |
| Generate report using | allure-maven@2.10.0 |

**Additional context**
Flaky tests are properly displayed when report is generated using allure 2.9.0

**Comment by PavloNikolenko** [comment]
Apparently PR #871 completely changed the way flaky tests are displayed in the report.
Correct me if I am wrong but it looks like `RetryPlugin` used to be responsible for marking the tests flaky if previous attempts were failed.
Now in #871 this responsibility was moved to the `HistoryPlugin` which completely ignores the results of `RetryPlugin` and instead looks at previous runs (vs previous attempts in the same run).

Can someone please clarify if this is intended behavior? At the moment there is no direct way to tell if one specific run had flaky tests or not.

@baev pinging you to get some visibility

**Comment by baev** [comment]
you are right, the behaviour was changed via #871 . At the moment it rely on history to calculate flaky tests. That seems to be more correct that in previous. 

I guess the solution should be to add additional marker for tests with retries.

**Comment by PavloNikolenko** [comment]
Thanks for a quick response.
I do find both flakiness by history and by retries useful. However, current check seems odd to me:

>     private boolean isFlaky(final List<HistoryItem> histories) {
>         if (histories.size() > 1 && histories.get(0).status == Status.FAILED) {
>             final List<Status> statuses = histories.subList(1, histories.size())
>                     .stream()
>                     .sorted(comparingByTime())
>                     .map(HistoryItem::getStatus)
>                     .collect(Collectors.toList());
>             return statuses.indexOf(Status.PASSED) < statuses.lastIndexOf(Status.FAILED)
>                     && statuses.indexOf(Status.PASSED) != -1;
>         }
>         return false;
>     }
This will mark the test as flaky only if it's latest result is failed. For example pass-fail-pass should be flaky as well. Don't you think?

I believe this check can be simplified to something like "mark the test flaky if it passed and failed (including broken?) at least once in the past 5 runs"?

**Comment by baev** [comment]
There are many different situations that can be used to mark test case as flaky

1) directly marked as flaky by user (why not)
2) has success rate (calculated over last X runs) < some threshold
3) has more then X different failures over last Y runs
4) sometimes has the same error over last X runs
5) has different failures in retries without code changes (some people can fix the bug and then build aggregated report)
6) has broken rate (calculated over last X runs) > some threshold

**Comment by baev** [comment]
I rather prefer implementing all the rules mentioned above instead of creating one magical "flaky" marker in order to make it more clear for end users. So having all rules implemented users can just review all the test cases with warnings and consider future actions such as mute test case, update it or delete it completely

**Comment by baev** [comment]
The current solution is contributed by company that marks flaky by it's own way. I can't say that I like it, but it looks better then the previous one, based on retries. But the problem is that we dropped old marker was used by users to find tests with retries. So at first we should add new "has retries" marker (that would be useful anyway), and then keep working on improving flaky tests algorithm

**Comment by PavloNikolenko** [comment]
I like the idea of "has retries" marker. How about displaying status icons for retries? Something like
![retries](https://user-images.githubusercontent.com/15851089/56928696-7dcc0880-6a8c-11e9-846a-b64dcd9c1370.png)
Maybe smaller icons and only last couple attempts (in case there are too many)?
